### DISCUSSION TITLE
How to set up a block as a heat source with constant power (unit: W/m^3)

### DISCUSSION THREAD

**Discussion by watercloud0** [idaholab/moose#23811] [Q&A General] [answered]
Hello,

How can I set a constant power in W/m^3 for a block object using HeatSource, or is there another way to set it?
I looked at this page link (https://mooseframework.inl.gov/moose/source/kernels/HeatSource.html#fde994ce-ae75-4dde-b484-8c843e4a4e38) and found the following example
  [. /source]
    type = HeatSource
    variable = temp
    block = 3
    value = 10.0
  [... /]
But what I'm confused about is whether the unit for value is W/m^3 or K?

Thanks a lot.

**Comment by GiudGiud** [comment, marked as answer]
Hello

the unit for a heat source is W/m3 as long as your mesh is in meters

Guillaume

**Reply by watercloud0** [reply]
Thanks for your explanation!
